### ISSUE TITLE
[Android] manually define adb path and skip check for local sdk installation

### ISSUE THREAD

**Issue by cs01** [wix/Detox#1579]
**Is your feature request related to a problem? Please describe.**

We are running a setup where we lease a remote android emulator and control it via adb (I think this is similar to how Genymotion works). In this setup, the local machine has adb but does not have a a full sdk installed. I am using the `android.attached` type since the emulator is already attached to adb.

When I try to use detox in this setup the error `$ANDROID_SDK_ROOT is not defined...` is thrown. If I go in and explicitly set the adb path in [detox's  source](https://github.com/wix/Detox/blob/dd34555bc2c23b24c16ad8e0b407c92115b0de8e/detox/src/devices/android/ADB.js#L12), I'm able to successfully run a Detox test since `getAndroidSDKPath()` isn't called. 

(I also have to run `adb reverse tcp:1234 tcp:1234` for the app to login to the detox server, but that works fine.)

**Describe the solution you'd like**
A flag or argument to define the path to `adb` instead of requiring the entire sdk to be installed.(https://github.com/wix/Detox/blob/dd34555bc2c23b24c16ad8e0b407c92115b0de8e/detox/src/devices/android/ADB.js#L12)

cc @zsol

**Comment by zsol** [comment]
on a related note, it would be nice if the device driver would send the appropriate `adb reverse` command in `prepare()` - we could

1. send down the url from `sessionConfig` [here](https://github.com/wix/Detox/blob/master/detox/src/devices/Device.js#L23)
2. parse the url and check if the port is already forwarded out
3. do `adb reverse` if not

**Comment by cs01** [comment]
IIUC the `android.attached` type means the device is already attached, and that `adb devices` will list an attached emulator with the device name give to detox. 

Whether the or not the emulator was created locally with a local sdk or remotely doesn't matter; only that it's already attached. Consequently, detox does not need to create or manage it and the following call is a no-op:

https://github.com/wix/Detox/blob/6a21e6cdf4dbd5746eea4dfd3ca6a546fae6706c/detox/src/devices/drivers/AttachedAndroidDriver.js#L9

In our setup, that call actually breaks the test because the machine with the attached emulator doesn't have the sdk installed on it. The emulator is running remotely, but adb is attached to the local machine and forwards commands to the remote emulator. For use to use detox in this setup, detox cannot make any calls that require the `ANDROID_SDK_ROOT` to exist. 

I modified detox locally and ran these tests, and it worked 🎉 , so I created a PR with these changes. I don't believe it will break anything because the `new Emulator()` call is a no-op and was just copy+pasted from the `EmulatorDriver` class. That class uses `this.emulator` for the following calls, none of which are necessary when using `android.attached`: `acquireFreeDevice`, `boot`, `shutdown`.
